synapse/handlers/sso.py

```diff
@@ -816,6 +851,15 @@ async def register_sso_user(self, request: Request, session_id: str) -> None:
             path=b"/",
         )
 
+        auth_result = {}
+        if session.terms_accepted_version:
+            # TODO: make this less awful.
+            auth_result[LoginType.TERMS] = True
+
+        await self._registration_handler.post_registration_actions(
+            user_id, auth_result, access_token=None
+        )
```

Comment:
Was this just not getting called before?

Reply:
indeed it wasn't. It didn't matter, because the only thing it does other than register consent is registering 3pids, which we do in `_register_mapped_user` anyway.

doing it this way feels horrible, because I'm basically having to pretend that we're doing a UIA registration here, but I don't want to get into refactoring all that right now. I don't really understand why `post_registration_actions` is separate from `register_user` anyway. (It seems to have arrived in #4682: any thoughts?)

Reply:
I think I probably did it like that because there was some logic/processing that could be done on the workers between the two calls, or something. I can't really remember if I'm honest.

I ha